metricbeat/module/kibana/stats/stats.go

```diff
@@ -44,20 +45,36 @@ var (
 // MetricSet type defines all fields of the MetricSet
 type MetricSet struct {
 	mb.BaseMetricSet
-	http *helper.HTTP
+	http         *helper.HTTP
+	xPackEnabled bool
 }
 
 // New create a new instance of the MetricSet
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	cfgwarn.Experimental("The kibana stats metricset is experimental")
 
+	config := kibana.DefaultConfig()
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	if config.XPackEnabled {
+		cfgwarn.Experimental("The experimental xpack.enabled flag in kibana/stats metricset is enabled.")
+	}
+
 	http, err := helper.NewHTTP(base)
 	if err != nil {
 		return nil, err
 	}
+
+	if config.BasePath != "" {
+		http.AddBasePath(config.BasePath)
```

Comment:
@ruflin Since Kibana URIs can have an optional basepath, I'm allowing users of the `kibana` module to set it in their config YAML. If it is set, I insert it into the URI using `http.AddBasePath` as shown here. Please see my implementation of the `AddBasePath` method (elsewhere in this PR) as well.

WDYT - is this config setting the right approach? If yes, is the implementation of the `AddBasePath` method alright?

Reply:
In general LGTM. See my comment above if this is only for dev purpose and if yes, if the host config change could work.